plugin-build/build.gradle.kts

```diff
@@ -61,6 +62,11 @@ dependencies {
     compileOnly(Libs.ASM)
     compileOnly(Libs.ASM_COMMONS)
 
+    implementation(kotlin("gradle-plugin-api"))
+    implementation("org.jetbrains.kotlin:kotlin-stdlib:${KotlinCompilerVersion.VERSION}")
+    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:${KotlinCompilerVersion.VERSION}")
```

Comment:
Also, I think these should be `compileOnly` following the AGP pattern we have, right? Otherwise we'll be pulling in our kotlin-plugin version a not use the one of the host